solutions/ac-GCDOnBlackboard.mdx

```diff
@@ -114,4 +114,62 @@ print(ans)
 ```
 
 </PySection>
+
+<JavaSection>
+```java
+import java.util.*;
+import java.io.*;
+
+public class gcdBoard {
+    static final int maxN = 100000 + 5;
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        int N = Integer.parseInt(br.readLine());
+        StringTokenizer st = new StringTokenizer(br.readLine());
+
+        int[] arr = new int[maxN];
+        for (int x = 1; x <= N; x++) {
+            arr[x] = Integer.parseInt(st.nextToken());
+        }
+
+        int[] prefix = new int[maxN];
+	// Prefix will find the GCD from the start
+        int[] suffix = new int[maxN];
+	// Suffix will find the GCD from the end
+
+        prefix[0] = 0;
+        suffix[N + 1] = 0;
+
+        for (int x = 1; x <= N; ++x) {
+		// Find GCD from first
+            prefix[x] = GCD(prefix[x - 1], arr[x]);
+        }
+        for (int x = N; x >= 1; --x) {
+		// Find GCD from second
+            suffix[x] = GCD(suffix[x + 1], arr[x]);
+        }
+
+        int ans = 1;
+
+        for (int x = 1; x <= N; ++x) {
+		// 'x' is the index we are going to remove
+            ans = Math.max(ans, GCD(prefix[x - 1], suffix[x + 1]));
+		// Removing index 'x is the same as finding the GCD of everything
+		// before it and after it
+        }
+
+        System.out.println(ans);
+    }
+
+    public static int gcd(int num1, int num2) {
+	// How to quickly find GCD
+        if (num2 != 0) {
+            return GCD(num2, num1 % num2);
```

Comment:
make sure to change these calls to lower case (there are also calls in the main method)